### PR TITLE
Ecs create server group

### PR DESCRIPTION
**Pull request by dkirillov**
@BrunoCarrier

### CODE REVIEW

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+
+public class CreateServerGroupAtomicOperation implements AtomicOperation<Void> {
+  private static final String BASE_PHASE = "CREATE_ECS_SERVER_GROUP";
+
+  private static final String IMAGE_ARN = "abcd";
```
[the remark this answers is not in the document]
Comment (dkirillov) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java]: Done, and yes the value should have been a docker registry (ECR in this case) URL.

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    AmazonECS ecs = amazonClientProvider.getAmazonEcs(description.getCredentialAccount(), credentials.getCredentialsProvider(), description.getRegion() != null ? description.getRegion() : REGION);
+    String familyName = getFamilyName();
+    TaskDefinition taskDefinition = registerTaskDefinition(ecs);
+
+    DeploymentConfiguration deploymentConfiguration = new DeploymentConfiguration();
+    deploymentConfiguration.setMinimumHealthyPercent(50);
+    deploymentConfiguration.setMaximumPercent(100);
+
+    LoadBalancer loadBalancer = new LoadBalancer();
+    loadBalancer.setContainerName(CONTAINER_NAME);
```
[the remark this answers is not in the document]
Comment (dkirillov) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java]: That's just setting the container name :) 

A couple lines below that is:
```
loadBalancer.setTargetGroupArn(targetGroupARN);
```

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+    return null;
+  }
+
+  private TaskDefinition registerTaskDefinition(AmazonECS ecs) {
+    KeyValuePair serverGroupEnv = new KeyValuePair();
+    serverGroupEnv.setName("server_group");
+    serverGroupEnv.setValue(description.getServerGroupVersion());
+
+    KeyValuePair cloudStackEnv = new KeyValuePair();
```
[the remark this answers is not in the document]
Comment (dkirillov) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java]: Done.
`CLOUD_DETAIL` added.
Key names have been capitalized.

**Comment by BrunoCarrier** [comment]
LGTM +2

**Pull request merged**